ddtrace/tracer/tracer.go

```diff
@@ -666,7 +666,14 @@ func (t *tracer) updateSampling(ctx ddtrace.SpanContext) {
 	if t.rulesSampling == nil || sctx.trace == nil || sctx.trace.root == nil {
 		return
 	}
+	sctx.mu.Lock()
+	defer sctx.mu.Unlock()
+	if sctx.trace.locked {
```

Comment:
Here we're locking the context's lock, not the trace's lock. Why is there a difference?

Reply:
That was a mistake, fixed.